go/store/prolly/tree/json_indexed_document.go

```diff
@@ -107,29 +106,40 @@ func getBytesFromIndexedJsonMap(ctx context.Context, m StaticJsonMap) (bytes []b
 	return bytes, err
 }
 
-// Lookup implements types.SearchableJSON
-func (i IndexedJsonDocument) Lookup(ctx context.Context, pathString string) (sql.JSONWrapper, error) {
-	if strings.Contains(pathString, "*") {
-		// Optimized lookup doesn't currently support wildcards. Fall back on an unoptimized approach.
-		// TODO: Optimized lookup on wildcards.
-		val, err := i.ToInterface()
-		if err != nil {
-			return nil, err
-		}
-		nonIndexedDoc := types.JSONDocument{Val: val}
-		return nonIndexedDoc.Lookup(ctx, pathString)
-	}
-	result, err := i.tryLookup(ctx, pathString)
-	if err == unknownLocationKeyError {
-		if sqlCtx, ok := ctx.(*sql.Context); ok {
-			sqlCtx.GetLogger().Warn(err)
+func tryWithFallback(
+	ctx context.Context,
+	i IndexedJsonDocument,
+	tryFunc func() error,
+	fallbackFunc func(document types.JSONDocument) error) error {
+	err := tryFunc()
+	if err == unknownLocationKeyError || err == unsupportedPathError {
+		if err == unknownLocationKeyError {
+			if sqlCtx, ok := ctx.(*sql.Context); ok {
+				sqlCtx.GetLogger().Warn(err)
```

Comment:
Do we already have tests that cover using unsupported path features and test that the fallback works correctly and that we log a session warning?